File: Backend/src/entities/ExerciseRef.ts
import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";

@Entity()
export class ExerciseRef {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    name: string;

    @Column()
    durationVSReps: boolean;

    @Column()
    weight: boolean;

    @Column()
    distance: boolean;

    @Column({ nullable: true }) //todo: NOT NULLABLE
    description?: string;

    @Column({ nullable: true }) //todo: NOT NULLABLE
    imagePath?: string;

    @Column({ default: false })
    isCustom: boolean;
}

File: MuscleMemo/app/src/main/java/com/cs346/musclememo/classes/ExerciseRef.kt
package com.cs346.musclememo.classes


data class ExerciseRef(
    val name: String = "Unknown Exercise",
    val id : Int = -1,
    val durationVSReps: Boolean = false,
    val weight: Boolean = false,
    val distance: Boolean = false,
    val isCustom: Boolean = false,
    val imagePath: String? = null,
    val description: String? = null
)